packages/grid/_modules_/grid/models/colDef/gridBooleanOperators.ts

```diff
@@ -1,20 +1,18 @@
 import { GridFilterInputBoolean } from '../../components/panel/filterPanel/GridFilterInputBoolean';
 import { GridFilterItem } from '../gridFilterItem';
 import { GridFilterOperator } from '../gridFilterOperator';
-import { GridColDef } from './gridColDef';
 
 export const getGridBooleanOperators: () => GridFilterOperator[] = () => [
   {
     value: 'is',
-    getApplyFilterFn: (filterItem: GridFilterItem, column: GridColDef) => {
+    getApplyFilterFn: (filterItem: GridFilterItem) => {
       if (!filterItem.columnField || !filterItem.value || !filterItem.operatorValue) {
         return null;
       }
 
       const valueAsBoolean = filterItem.value === 'true';
-      return (params): boolean => {
-        const rowValue = column.valueGetter ? column.valueGetter(params) : params.value;
-        return Boolean(rowValue) === valueAsBoolean;
+      return ({ value }): boolean => {
+        return Boolean(value) === valueAsBoolean;
```

Comment:
Should we change the type of `params` to not allow it to be passed to `valueGetter`? This is just to prevent us and users from making the same mistake in the future.

Reply:
I don't think so, as the issue was within the filters and not in the user's demo.

Reply:
@dtassone but developer can't write custom operators and make the same mistake?

Reply:
What trap? I'm not sure I understand.
there is a type 
```tsx
export interface GridFilterOperator {
  label?: string;
  value: string;
  getApplyFilterFn: (
    filterItem: GridFilterItem,
    column: any,
  ) => null | ((params: GridCellParams) => boolean);
  InputComponent: React.JSXElementConstructor<GridFilterInputValueProps>;
  InputComponentProps?: Record<string, any>;
}

```

Reply:
```js
{
  field: "country", // field exists in data
  headerName: "Country",
  valueGetter: (cellParams) => cellParams.value.name,
  flex: 0.5
}
```

```js
{
  field: "country2", // field doesn't exist in data
  headerName: "Country",
  valueGetter: (cellParams) => cellParams.row.country.name,
  flex: 0.5
}
```

The trap is that if the `valueGetter` relays on the value (first example), it can't be called inside the filter operator because `cellParams.value` will be already the resolved value (a string). If `valueGetter` relays on `row` (second example) there's no trap, but we don't know a priori what is used.

Reply:
we just have to know that cellParams consider valueGetter, which should be expected as the value wouldn't be correct otherwise.